amlb/benchmarks/openml.py

```diff
@@ -20,7 +21,13 @@ def is_openml_benchmark(benchmark: str) -> bool:
 
 def load_oml_benchmark(benchmark: str) -> tuple[str, str | None, list[Namespace]]:
     """Loads benchmark defined by openml suite or task, from openml/s/X or openml/t/Y."""
-    domain, oml_type, oml_id = benchmark.split("/")
+    domain, oml_type, oml_id_str = benchmark.split("/")
+    try:
+        oml_id = int(oml_id_str)
+    except ValueError:
+        raise ValueError(
+            f"Could not convert OpenML id {oml_id_str!r} in {benchmark!r} to integer."
+        )
```

Comment:
The reason it was sometimes `str` and sometimes not was because when specified as an openml benchmark (openml/{s,t}/{id} it was passed as str. But when loaded from a file the ids were conventionally defined as integers.